### PR TITLE
Raise an error when index name is invalid

### PR DESCRIPTION
**Pull request by Thibaut**
Prevents this kind of thing:

```ruby
change_table :foo do |m|
  m.add_index [:bar], name: 'bar'
end
```

```sql
create index `{:name=>"bar"}` on `lhmn_foo` (`bar`)
```

@arthurnn @camilo 
cc @sroysen @SingerWang

### CODE REVIEW

**Comment by Thibaut** [comment]
@arthurnn are you rebasing `Shopify/lhm` soon or should I merge this here?

**Comment by arthurnn** [comment]
Merge this for now.. I will rebase when I can .=(

File: lib/lhm/migrator.rb [hunk 1]
```diff
@@ -205,5 +206,11 @@ def index_ddl(cols, unique = nil, index_name = nil)
     def order_column
       @options[:order_column] || @origin.pk
     end
+
+    def assert_valid_idx_name(index_name)
+      if index_name && !(index_name.is_a?(String) || index_name.is_a?(Symbol))
+        raise ArgumentError, "index_name must be a string or symbol"
```
Comment (camilo) [inline lib/lhm/migrator.rb]: now that you are here, might as well make sure the index is smaller than the max size

Comment (SingerWang) [inline lib/lhm/migrator.rb]: http://dev.mysql.com/doc/refman/5.6/en/identifiers.html

64 characters ;p

Comment (arthurnn) [inline lib/lhm/migrator.rb]: please do in another commit and send upstream !

Comment (Thibaut) [inline lib/lhm/migrator.rb]: > now that you are here, might as well make sure the index is smaller than the max size

Do we want to check that in Ruby? What if the max size changes in a future version of MySQL?

If the name is too long then the migration will fail.

Comment (arthurnn) [inline lib/lhm/migrator.rb]: good point!

Comment (camilo) [inline lib/lhm/migrator.rb]: > Do we want to check that in Ruby?

Maybe, you are right about the migration failing.  I can be very annoying when indexes are derived from the column names and you have to wait after deploying to realize shit's broken. On the other hand ppl should test locally.

Comment (Thibaut) [inline lib/lhm/migrator.rb]: > you have to wait after deploying to realize shit's broken

Hum, wat? If the migration fails then it cannot be deployed. Shouldn't even pass CI.

On the other hand the exception I'm adding above is preventing a valid but bad migration from going out. If the index is too long then the LHM should blow up at run time, which is good enough IMHO.

Comment (camilo) [inline lib/lhm/migrator.rb]: hokay

File: lib/lhm/migrator.rb [hunk 2]
```diff
@@ -196,6 +196,7 @@ def destination_read
     end
 
     def index_ddl(cols, unique = nil, index_name = nil)
+      assert_valid_idx_name(index_name)
```
Comment (camilo) [inline lib/lhm/migrator.rb]: super duper nitpick, why not  `validate_*` instead of `assert_*` I find that assert_ has a testing flavor in rubby

**Comment by camilo** [comment]
:sake: it besides my naming nitpick.

File: lib/lhm/migrator.rb (hunk 2 above)
Comment (Thibaut) [inline lib/lhm/migrator.rb]: @camilo since this is already merged upstream I'd rather leave it as in. `assert_*` is a thing in Rails to validate arguments, see [`Hash#assert_valid_keys`](http://devdocs.io/rails/hash#method-i-assert_valid_keys).

Comment (camilo) [inline lib/lhm/migrator.rb]: sure

**Comment by Thibaut** [comment]
@camilo I don't have write access to this repo, can you merge?
